plugin/pkg/log/log.go

```diff
@@ -13,11 +13,33 @@ import (
 	"io/ioutil"
 	golog "log"
 	"os"
+	"sync"
 	"time"
 )
 
-// D controls whether we should output debug logs. If true, we do.
-var D bool
+// D controls whether we should output debug logs. If true, we do, once set
+// it can not be unset.
+var D = &d{}
+
+type d struct {
+	on bool
+	sync.RWMutex
+}
+
+// Set sets d to true.
+func (d *d) Set() {
+	d.Lock()
+	d.on = true
+	d.Unlock()
+}
+
+// Value return the boolean value of d.
+func (d *d) Value() bool {
```

Comment:
Not sure how much it matters but:
```
type d int32
...
return atomic.LoadInt32((*int32)(&d)) > 0
```
is probably faster.

Reply:
Missed your comment. I rather use the abstraction from the go library instead of doing the atomics myself.

Reply:
:+1: